src/utils/openmc/universe.py

```diff
@@ -1058,7 +1073,13 @@ def set_universes(self, universes):
         # The sub-lists are ordered from outermost ring to innermost ring.
         # The Universes within each sub-list are ordered from the "top" in a
         # clockwise fashion.
-        self.set_num_rings(universes.shape[0])
+        # TODO: Does this need some work to handle numpy arrays?  I'm not sure
```

Comment:
Actually, I think that NumPy arrays will work just fine. You can take a "ragged" 2D list and create a NumPy array from it, though it will create a 1D array where each element is itself a list. See an example below:

```
import numpy
a = [[1]*2, [3]*5, 4]
b = np.array(a)
print(b)
[[1, 1] [3, 3, 3, 3, 3] 4]
```